src/main/java/com/kodilla/ecommercee/service/DbUserService.java

```diff
@@ -0,0 +1,71 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.TokenNotFoundException;
+import com.kodilla.ecommercee.domain.User;
+import com.kodilla.ecommercee.domain.UserNotFoundException;
+import com.kodilla.ecommercee.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.security.SecureRandom;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class DbUserService {
+    @Autowired
+    private UserRepository userRepository;
+
+    public List<User> getAllUsers() {
+        return userRepository.findAll();
+    }
+
+    public Optional<User> getUser(final Long id) throws UserNotFoundException {
+        return userRepository.findById(id);
+    }
+
+    public void deleteUser(final Long id) {
+        if (userRepository.existsById(id)) {
+            userRepository.deleteById(id);
+        }
+    }
+
+    public User updateUser(final Long id, User user) throws UserNotFoundException {
+        User toUpdate = userRepository.findById(id).orElseThrow(UserNotFoundException::new);
+        toUpdate.setUsername(user.getUsername());
+        toUpdate.setToken(user.getToken());
+        toUpdate.setIsBlocked(user.getIsBlocked());
+        return userRepository.save(toUpdate);
+    }
+
+    public User saveUser(final User user) {
+        return userRepository.save(user);
+    }
+
+    public void blockUser(final Long id) {
+        if (userRepository.existsById(id)) {
+            User userToBlock = userRepository.getOne(id);
+            if (userToBlock.getIsBlocked().equals(false)) {
+                userToBlock.setIsBlocked(true);
+            } else {
+                userToBlock.setIsBlocked(false);
+            }
```

Comment:
else nie jest potrzebny

Reply:
blockUser ma działanie podwójne - jeśli użytkownik nie jest zablokowany, to ma zostać ustawiony status na zablokowanie, a jeśli użytkownik jest zablokowany, to wówczas ma za zadanie odblokować usera. Jak dla mnie logic.

Reply:
racja :)